Library/Homebrew/cmd/update.sh

```diff
@@ -378,19 +378,34 @@ EOS
   for DIR in "$HOMEBREW_REPOSITORY" "$HOMEBREW_LIBRARY"/Taps/*/*
   do
     [[ -d "$DIR/.git" ]] || continue
+    cd "$DIR" || continue
+
+    TAP_VAR="$(repo_var "$DIR")"
+    declare PREFETCH_REVISION"$TAP_VAR"="$(read_current_revision)"
+
     [[ -n "$SKIP_FETCH_BREW_REPOSITORY" && "$DIR" = "$HOMEBREW_REPOSITORY" ]] && continue
     [[ -n "$SKIP_FETCH_CORE_REPOSITORY" && "$DIR" = "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core" ]] && continue
-    cd "$DIR" || continue
+
+    # The upstream repository's default branch may not be master;
+    # check refs/remotes/origin/HEAD to see what the default
+    # origin branch name is, and use that. If not set, fall back to "master".
     UPSTREAM_BRANCH="$(upstream_branch)"
+    declare UPSTREAM_BRANCH"$TAP_VAR"="$UPSTREAM_BRANCH"
+
     # the refspec ensures that the default upstream branch gets updated
     (
       if [[ -n "$HOMEBREW_UPDATE_PREINSTALL" ]]
       then
         # Skip taps checked/fetched recently
-        [[ -n "$(find "$DIR/.git/FETCH_HEAD" -type f -mmin -1 2>/dev/null)" ]] && exit
-        # Skip taps without formulae.
-        FORMULAE="$(find "$DIR" -maxdepth 1 \( -name "*.rb" -or -name Formula -or -name HomebrewFormula \) -print -quit)"
-        [[ -z "$FORMULAE" ]] && exit
+        [[ -n "$(find "$DIR/.git/FETCH_HEAD" -type f -mtime -60s 2>/dev/null)" ]] && exit
```

Comment:
That seems simplest until there's some more formal config file and/or mechanism for brew generally like a `brew get foo` and `brew set foo=bar` or however that ends up working. Alternatively, it could be exposed now with a specific command like `brew analytics` is for the on/off value.